### PR TITLE
PLT-250 Fix post list not resizing properly for file previews.

### PR DESCRIPTION
**Pull request by jwilander**
I also fixed some style issues.

My bug fixes are on:
- new line 58
- new line 246

### CODE REVIEW

**Comment by coreyhulen** [comment]
+1

File: web/react/components/create_post.jsx [hunk 1]
```diff
@@ -237,8 +243,8 @@ export default class CreatePost extends React.Component {
         Utils.setCaretPosition(React.findDOMNode(this.refs.textbox.refs.message), newText.length);
     }
     removePreview(id) {
-        let previews = this.state.previews;
-        let uploadsInProgress = this.state.uploadsInProgress;
+        const previews = $.extend(true, [], this.state.previews);
```
Comment (crspeller) [inline web/react/components/create_post.jsx]: Possibly use ES6 `Object.assign()` here.

Comment (jwilander) [inline web/react/components/create_post.jsx]: Can that be used for arrays? I looked at the documentation quickly and it didn't mention arrays

Comment (jwilander) [inline web/react/components/create_post.jsx]: Yup, it does. Will switch to use it

**Comment by jwilander** [comment]
@crspeller Ready for re-review

**Comment by crspeller** [comment]
+1

**Comment by crspeller** [comment]
Merge when travis passes.

**Pull request merged**